tools/codegen/dest/lazy_ir.py

```diff
@@ -10,30 +10,36 @@
 from tools.codegen.dest.lazy_ts_lowering import ts_lowering_body
 
 
-def node_ctor_inputs(func: LazyIrSchema) -> str:
+def node_ctor_arg_rvalue_string(arg: NamedCType) -> str:
     """
-    Produce a formatted string with the arguments as passed into the constructor of a node class.
+    Given a NamedCType from a lazy IR schema,
+    generate a c++ string for materializing an rvalue of that arg for passing into
+    a lazy Node constructor.
     """
-    node_ctor_values = []
-    for arg in func.filtered_types():
-        if isValueType(arg.type):
-            if isinstance(arg.type, BaseCType):
-                node_ctor_values.append(f"lazy_{arg.name}.GetIrValue()")
-            elif isinstance(arg.type, OptionalCType):
-                node_ctor_values.append(
-                    f"lazy_{arg.name} ? "
-                    f"c10::make_optional(lazy_{arg.name}->GetIrValue()) : "
-                    "c10::nullopt")
-            else:
-                raise AssertionError("TODO not sure if there are other valid types to handle here")
+    if isValueType(arg.type):
+        if isinstance(arg.type, BaseCType):
+            return f"lazy_{arg.name}.GetIrValue()"
+        elif isinstance(arg.type, OptionalCType):
+            return f"lazy_{arg.name} ? " \
+                   f"c10::make_optional(lazy_{arg.name}->GetIrValue()) : " \
+                   "c10::nullopt"
         else:
-            if isinstance(arg.type, VectorCType) and isinstance(arg.type.elem, BaseCType):
-                node_ctor_values.append(f"lazy_tensors::util::ToVector<{arg.type.elem.type}>({arg.name})")
-            else:
-                node_ctor_values.append(f"{arg.name}")
+            raise AssertionError("TODO not sure if there are other valid types to handle here")
+    else:
+        if isinstance(arg.type, VectorCType) and isinstance(arg.type.elem, BaseCType):
+            return f"lazy_tensors::util::ToVector<{arg.type.elem.type}>({arg.name})"
+        elif (isinstance(arg.type, OptionalCType) and
+                isinstance(arg.type.elem, VectorCType)):
+            return f"lazy_tensors::util::ToOptionalVector<{arg.type.elem.elem.type}>({arg.name})"
+        else:
+            return f"{arg.name}"
 
-    node_ctor_inputs_str = ",\n                              ".join(node_ctor_values)
-    return node_ctor_inputs_str
+def node_ctor_inputs(func: LazyIrSchema) -> str:
```

Comment:
I think it would maybe be more readable to couple this with the actual call-site. Since you have two tightly coupled pieces of code (the arguments to the node constructor, and the name of the constructor itself) that are very non-local. Maybe this should just return the entire string:
```
return f'torch::lazy::MakeNode<ir::ops::{func.node_name}>({node_ctr_args})'
```

Reply:
yea that's a good point. i'll clean it up.